File: ex07/ex07_01/ex07_01.py
def main():
   file_name = "../data/ex07_01.txt"
   print("Enter information: ")
   pf = open(file_name, 'w')
   if pf is None:
       print("Error open file")
       return
   while True:
       text = input()
       if text == "":
           pf.close()
           print("Data saved to " + file_name)
           break
       pf.write(text + "\n")

if __name__ == "__main__":
    main()

File: ex07/ex07_01/ex07_01.c
#include <stdio.h>

int main() {
  char file_name[] = "../data/ex07_01.txt";
  char text[1024];
  FILE* pf;
  pf = fopen(file_name, "w");

  if (pf == NULL) {
    printf("Error open file");
    return 1;
  }
  printf("Enter information: \n");
  while(1) {
    fgets(text, sizeof(text), stdin);
    if (text[0] == '\n') {
      printf("Data saved to %s\n", file_name);
      break;
    }
    fprintf(pf, "%s", text);
  }
  fclose(pf);
  return 0;
}